.jenkinsci/bindings.groovy

```diff
@@ -1,27 +1,69 @@
 #!/usr/bin/env groovy
 
-def doBindings() {
-  def cmake_options = ""
-  if (params.JavaBindings) {
-    cmake_options += " -DSWIG_JAVA=ON "
-  }
-  if (params.PythonBindings) {
-    cmake_options += " -DSWIG_PYTHON=ON "
-  }
-  // In case language specific options were not set,
-  // build for each language
-  if (!params.JavaBindings && !params.PythonBindings) {
-    cmake_options += " -DSWIG_JAVA=ON -DSWIG_PYTHON=ON "
-  }
+def doJavaBindings(buildType=Release) {
+  def currentPath = sh(script: "pwd", returnStdout: true).trim()
+  def commit = env.GIT_COMMIT
+  def artifactsPath = sprintf('%1$s/java-bindings-%2$s-%3$s-%4$s.zip', 
+    [currentPath, buildType, sh(script: 'date "+%Y%m%d"', returnStdout: true).trim(), commit.substring(0,6)])
   sh """
     cmake \
       -H. \
       -Bbuild \
-      -DCMAKE_BUILD_TYPE=Release \
-      ${cmake_options}
+      -DCMAKE_BUILD_TYPE=$buildType \
+      -DSWIG_JAVA=ON
+  """
+  sh "cd build; make -j${params.PARALLELISM} irohajava"
+  sh "zip -j $artifactsPath build/shared_model/bindings/*.java build/shared_model/bindings/libirohajava.so"
+  sh "cp $artifactsPath /tmp/bindings-artifact"
+  return artifactsPath
+}
+
+def doPythonBindings(buildType=Release) {
+  def currentPath = sh(script: "pwd", returnStdout: true).trim()
+  def commit = env.GIT_COMMIT
+  def artifactsPath = sprintf('%1$s/python-bindings-%2$s-%3$s-%4$s.zip', 
+    [currentPath, buildType, sh(script: 'date "+%Y%m%d"', returnStdout: true).trim(), commit.substring(0,6)])
+  sh """
+    cmake \
+      -H. \
+      -Bbuild \
+      -DCMAKE_BUILD_TYPE=$buildType \
+      -DSWIG_PYTHON=ON \
+      -DSUPPORT_PYTHON2=ON
   """
   sh "cmake --build build --target python_tests"
-  sh "cd build; make -j${params.PARALLELISM} irohajava irohapy"
+  sh "cd build; make -j${params.PARALLELISM} irohapy"
+  sh "zip -j $artifactsPath build/shared_model/bindings/*.py build/shared_model/bindings/*.so"
+  sh "cp $artifactsPath /tmp/bindings-artifact"
+  return artifactsPath
+}
+
+def doAndroidBindings(abiVersion) {
+  def currentPath = sh(script: "pwd", returnStdout: true).trim()
+  def commit = env.GIT_COMMIT
+  def artifactsPath = sprintf('%1$s/android-bindings-%2$s-%3$s-%4$s-%5$s-%6$s.zip', 
+    [currentPath, "\$PLATFORM", abiVersion, "\$BUILD_TYPE_A", sh(script: 'date "+%Y%m%d"', returnStdout: true).trim(), commit.substring(0,6)])
+  sh """
+    (cd /iroha; git init; git remote add origin https://github.com/hyperledger/iroha.git; \
+    git fetch --depth 1 origin develop; git checkout -t origin/develop)
+  """
+  sh """
+    . /entrypoint.sh; \
+    sed -i.bak "s~find_package(JNI REQUIRED)~#find_package(JNI REQUIRED)~" /iroha/shared_model/bindings/CMakeLists.txt; \
+    sed -i.bak "s~# the include path to jni.h and jni_md.h~SET(CMAKE_SWIG_FLAGS \\\${CMAKE_SWIG_FLAGS} -package \${PACKAGE})~" /iroha/shared_model/bindings/CMakeLists.txt; \
```

Comment:
This doesn't work - ```-package``` should be set before ```swig_add_library``` call (that's actually a bug in the current development branch). My suggestion is to use command
```
sed -i.bak "s~find_package(JNI REQUIRED)~SET(CMAKE_SWIG_FLAGS \${CMAKE_SWIG_FLAGS} -package ${PACKAGE})~" /iroha/shared_model/bindings/CMakeLists.txt
```
instead of this one and previous.